### PR TITLE
Update google test release-1.12.1

### PR DESCRIPTION
**Pull request by blowekamp**
## PR Checklist
- [x] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [x] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,19 +1,21 @@
 # Note: CMake support is community-based. The maintainers do not use CMake
 # internally.
 
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.5)
```
Comment (dzenanz) [inline CMakeLists.txt]: Oh, they finally moved to CMake 3.

**Review by dzenanz — APPROVED** [review]

**Review by hjmjohnson — APPROVED** [review]

**Comment by seanm** [comment]
This googletest update broke old clang.  Patch here:

https://github.com/google/googletest/pull/4024

**Comment by seanm** [comment]
@dzenanz seems like google does not like our little 2 line change... shall we just make the 2 line change in ITK ourselves then?

**Comment by dzenanz** [comment]
Didn't https://github.com/google/googletest/commit/a9b2f0495cabb1ff01b51bcd0625cb1fd82da4ce fix it?

**Pull request merged**